ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/lexer/LexerDirectoryProcessor.java

```diff
@@ -0,0 +1,38 @@
+package edu.usb.argos.ASTProcessor.lexer;
+
+import org.antlr.v4.runtime.CommonTokenStream;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public class LexerDirectoryProcessor {
```

Comment:
Instead of concrete implementations, it implements a generic interface to improve flexibility, as requested by the trainer.

Reply:
Done ✅